### ISSUE TITLE
`variable.debug_print()` when `variable.data` is `None` **bug**

### ISSUE THREAD

**Issue by EmbraceLife** [chainer/chainer#3994]
**environment:** 
MacOS Higher Sierra 10.13.1
chainer version 3.0.0
numpy version 1.13.3

**code to run to cause the error**
```python
import chainer
v = chainer.Variable()
v.debug_print()
```
**error message**
`TypeError: unsupported operand type(s) for /: 'NoneType' and 'int'`

**source of error located** 
https://github.com/chainer/chainer/blob/master/chainer/variable.py#L524 
`xp.mean(self.data)` refers to `numpy.mean(v.data)`, referring to `numpy.mean(None)`, which cause the same error message below
`TypeError: unsupported operand type(s) for /: 'NoneType' and 'int'`

**a solution proposed**
```python
if self.data is not None: 
   stats = stats_msg.format(float(xp.mean(self.data)),
                                     float(xp.std(self.data)))
else:
   stats = "None"

return msg.format(summary=self.summary(),
                          grad=grad, shape=self.data.shape if self.data is not None else None,
                          background=type(self.data) if self.data is not None else None,
                          dtype=self.data.dtype if self.data is not None else None, device=device,
                          stats=stats)
```

**Comment by hvy** [comment]
@EmbraceLife, thanks for pointing this out! I'll make a quick fix.

**Comment by kmaehashi** [comment]
This issue will be fixed in the next release (#4018).
Thanks again for reporting!
